library/src/lib/calendar/calendar.component.ts

```diff
@@ -614,7 +614,23 @@ export class CalendarComponent implements OnInit, OnDestroy, AfterViewChecked, C
     }
 
     setCurrentMonth(month: number) {
+        // get the current date of the month
+        const currentDate = this.date.getDate();
+        // get the number of days in the new month
+        const daysInNewMonth = new Date(this.date.getFullYear(), month + 1, 0).getDate();
+        /*
+         if the currentDate > daysInNewMonth, set the date to the first for now, to prevent skipping a month
+         in the event that the currentDate is 31 and the next month has 30 days
+         */
+        if (currentDate > daysInNewMonth) {
+            this.date.setDate(1);
```

Comment:
Does this go to the next month?

Reply:
No, it sets the date for the current month to 1, but sets it back later

Reply:
hmm not sure how friendly this for ux

Reply:
This is all happening internally, not with the actual ngModel the user sees

Reply:
Yeah I know, but so if the user is on the 31st and clicks next, it goes to date(1) of the same month that was currently displayed? Or the next month?

Reply:
The month that is currently displayed, but I don't think this really matters because it reverts back a few lines later

Reply:
Can you describe the PR in the description? And also exactly the flow of the bug? Like for instance, if the user presses the next month button, this and this happens.

Reply:
Added a little description.  The original ticket is linked in the description with a gif demonstration